runtime/lua/vim/lsp/buf.lua

```diff
@@ -61,11 +61,16 @@ function M.hover(config)
     local results1 = {} --- @type table<integer,lsp.Hover>
 
     for client_id, resp in pairs(results) do
+      --- @type lsp.ResponseError, lsp.Hover
       local err, result = resp.err, resp.result
       if err then
         lsp.log.error(err.code, err.message)
       elseif result then
-        results1[client_id] = result
+        local contents = result.contents
+        -- Ensure that there is some content to display
+        if type(contents) == 'string' or contents.kind or contents.language or contents[1] then
```

Comment:
A comment explaining that this corresponds to checking that `contents` is a `MarkedString | MarkedString[] | MarkupContent` would be appreciated.

Reply:
I've simplified the check to just make sure `#contents > 0` (i.e. it is not an empty table or an empty string)